website/docs/r/network_pool.html.markdown

```diff
@@ -0,0 +1,245 @@
+  parent        = data.vcd_vcenter.vc1.name
+}
+
+output "tzs" {
+  value = data.vcd_resource_list.tz.list
+}
+
+output "pgs" {
+  value = data.vcd_resource_list.pg.list
+}
+
+output "ds" {
+  value = data.vcd_resource_list.ds.list
+}
+```
+
+-> Note: the lists provided as `vcd_resource_list` output are volatile: they only exist for items that have not been used
+in a network pool. Once they have been assigned, they cease to be shown. As such, it is not a good idea to use
+`vcd_resource_list` as direct source for one or more network pools: at the first `plan`, terraform would propose
+to remove the network pool, as the element is not shown in the list anymore.
+
+## Example Usage 5 - nameless backing retrieval
+
+If we don't have preference about which of the elements we will use as backing for the network pool, we could let
+the system pick the first available. This could be a good idea when we know that there is only one element available, or 
+we know that all elements have similar capabilities.
+If we are in these circumstances, we could avoid some details and skip the definition of the backing elements, but we
+must specify the value of `backing_components_use_constraint` as either `use-when-only-one` or `use-first-available`.
+
+```hcl
+data "vcd_nsxt_manager" "mgr" {
+  name = "nsxManager1"
+}
+
+data "vcd_vcenter" "vc1" {
+  name = "vc1"
+}
+
+resource "vcd_network_pool" "npool-1" {
+  name                = "new-network-pool"
+  description         = "network pool without explicit transport zone"
+  network_provider_id = data.vcd_vcenter.vc1.id
+  type                = "PORTGROUP_BACKED"
+
+  backing_components_use_constraint = "use-when-only-one"
+
+  backing {
+  }
+}
+
+resource "vcd_network_pool" "npool-2" {
+  name                = "new-network-pool"
+  description         = "network pool without explicit transport zone"
+  network_provider_id = data.vcd_nsxt_manager.mgr.id
+  type                = "GENEVE"
+
+  backing_components_use_constraint = "use-first-available"
+
+  backing {
+  }
+}
+```
+
+The system will pick the first (or the only) available transport zone, or fail if none was available. The name of the used transport
+zone will be shown if we use an `output` for the network pool.
+
+```hcl
+output "pool1" {
+  value = vcd_network_pool.npool1.backing
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) Unique name of network pool
+* `type` - (Required) Type of the network pool (one of `GENEVE`, `VLAN`, `PORTGROUP_BACKED`)
+* `network_provider_id` - (Required) Id of the network provider (either vCenter or NSX-T manager)
+* `description` - (Optional) Description of the network pool
+* `backing_components_use_constraint` - (Optional) Define how the backing components are considered. It should be one of the following:
+    * `use-explicit-name` (default) The backing components must be named explicitly;
+    * `use-when-only-one` The unnamed backing component will be used if there is only one available;
```

Comment:
Doc polishing: what does it mean "unnamed" in this context, is it "non-user-defined in the backing block"? It would be good to use some more specific wording.

Reply:
rephrased